cms/serializers_test.py

```diff
@@ -28,3 +35,84 @@ def test_serialize_course_page(mocker, mock_context):
         },
     )
     patched_get_wagtail_src.assert_called_once_with(course_page.feature_image)
+
+
+def test_serialize_course_page_with_flex_price_with_program_fk(mocker, mock_context):
+    """
+    Tests course page serialization with Financial Assistance form which has a fk relationship to
+    a program, but no parent-child relationship with any course or program.
+    """
+    fake_image_src = "http://example.com/my.img"
+    mocker.patch("cms.serializers.get_wagtail_img_src", return_value=fake_image_src)
+
+    program = ProgramFactory()
+    program_page = ProgramPageFactory(program=program)
+    financial_assistance_form = FlexiblePricingFormFactory(
+        selected_program_id=program.id, parent=None
+    )
+    course = CourseFactory(program=program)
+    course_page = CoursePageFactory(course=course)
+
+    data = CoursePageSerializer(instance=course_page).data
+    assert_drf_json_equal(
+        data,
+        {
+            "feature_image_src": fake_image_src,
+            "page_url": None,
+            "financial_assistance_form_url": f"{program_page.get_url()}{financial_assistance_form.slug}/",
```

Comment:
See earlier comment about using get_url here (and below, on lines 89 and 116). Caveat: sometimes that doesn't work in a test suite - if it doesn't, this is fine as is.